src/int_allocator.rs

```diff
@@ -29,62 +37,66 @@ impl IntAllocator {
     pub fn new() -> Self {
         let mut r = IntAllocator {
             u8_vec: Vec::new(),
-            node_vec: Vec::new(),
+            pair_vec: Vec::new(),
+            atom_vec: Vec::new(),
         };
         r.u8_vec.reserve(1024 * 1024);
+        r.atom_vec.reserve(256);
```

Comment:
Oh, so the atoms and pairs are held on two separate stacks.

Reply:
yes, that's the core of this patch